code/js/modules/BaseController.js

```diff
@@ -214,7 +209,12 @@
       song: this.getSongData(this.selectors.song),
       artist: this.getSongData(this.selectors.artist),
       isPlaying: this.isPlaying(),
-      siteName: this.siteName
+      siteName: this.siteName,
+      canDislike: !!(this.selectors.dislike && this.doc().querySelector(this.selectors.dislike)),
+      canPlayPrev: !!(this.selectors.playPrev && this.doc().querySelector(this.selectors.playPrev)),
+      canPlayPause: !!((this.selectors.playPause && this.doc().querySelector(this.selectors.playPause)) || (this.selectors.play && this.doc().querySelector(this.selectors.play)) || (this.selectors.pause && this.doc().querySelector(this.selectors.pause))),
```

Comment:
That logic makes sense. Like I mentioned before, every player has at a minimum play/pause functionality so it follows that if playPause does not exist then neither does the player, good idea.